### PR TITLE
[SIMPLE_FORMS] feat: build out submission archiver service object

### PR DESCRIPTION
**Pull request by pennja**
## Summary

- This work creates the submission archiver service object
- This work also adds RSpec unit test coverage

## Related issue(s)

- [Create PoC for adding filled in PDFs to an S3 bucket](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1633)

## Testing done

- [x] *New code is covered by unit tests*

## Requested Feedback

Any

### CODE REVIEW

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          submission: nil             # The FormSubmission object representing the original data payload submitted
+        }
+      end
+
+      def build_submission_archive(**)
+        SubmissionArchiveBuilder.new(**).run
+      end
+
+      def upload_temp_folder_to_s3
+        Find.find(temp_directory_path) do |path|
```
Comment (Thrillberg) [inline modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb:74]: I'm not sure I understand what is going on here. Are we unable to provide the `relative_path` up front? Why do we need to use `Find.find`?

**Review by Thrillberg — DISMISSED** [review]
Looks great! Just one question, out of curiosity mostly. 👍

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb (hunk 1 above)
Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb:74]: Oops, this was probably a carry over from the original script I built this off of. I'll update it. Good catch!

Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb]: Aaaactually, it wasn't a carryover and I just remembered why I did this. This logic iterates through all files within the directory and explicitly reads each one into S3. I went ahead and made a slight change to make it more explicit.

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+      def build_submission_archive(**)
+        SubmissionArchiveBuilder.new(**).run
+      end
+
+      def upload_temp_folder_to_s3
+        Dir.glob("#{temp_directory_path}/**/*").each do |path|
+          next if File.directory?(path)
+
+          File.open(path, 'rb') { |file| save_file_to_s3(file.read) }
```
Comment (Thrillberg) [inline modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb:77]: Will this retain whatever directory structure you have? Or is it all just dumped into one folder?

**Review by Thrillberg — APPROVED** [review]
Looks great, nice work!

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb (hunk 2 above)
Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archiver.rb:77]: There shouldn't be any directory structure within this directory. It should just be a collection of files.

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**